src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateFromFile.Tests.cs

```diff
@@ -868,5 +869,40 @@ public void ReusingNames_Windows(string name)
             }
         }
 
+        /// <summary>
+        /// Test that we can map special character devices on Unix.
+        /// </summary>
+        [ConditionalTheory]
+        [InlineData(MemoryMappedFileAccess.Read)]
+        [InlineData(MemoryMappedFileAccess.ReadWrite)]
+        [InlineData(MemoryMappedFileAccess.Write)]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public void OpenCharacterDevice(MemoryMappedFileAccess access)
+        {
+            string device = "/dev/zero";
+            if (!File.Exists(device))
+            {
+                throw new SkipTestException($"'{device}' is not available.");
+            }
+
+            long viewCapacity = 0xFF;
+
+            using (FileStream fs = new FileStream(device, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite))
+            using (MemoryMappedFile memMap = MemoryMappedFile.CreateFromFile(fs, null, viewCapacity, access, HandleInheritability.None, false))
```

Comment:
Do we have unit tests that throw when `CreateNew` is called with `access = MemoryMappedFileAccess.Write`? Because if we do, they should fail with the current changes (see my comments above).

Reply:
We have some tests for invalid input but if 'mapName' is specific we expect to throw PNSP and that may come before getting to platform code.